### ISSUE TITLE
HLS seems to require a library stanza in the cabal file

### ISSUE THREAD

**Issue by Rotaerk** [haskell/haskell-language-server#3932]
## Environment

OS: NixOS
GHC Version: 9.4.8
Cabal Version: 3.10.2.1
HLS Version: 2.4.0.0

Everything is installed via a nix flake.  Here is my project at the time of issue submission: https://github.com/Rotaerk/vulkanSandbox/tree/85d04fd4a834000cb47a7992207f8c74c5028428

## What's wrong?

Within the repository root, I run `nix develop`, which brings cabal, ghc, and HLS among other things into my path.  I navigate into the VulkanSandbox directory, and run `haskell-language-server-wrapper`.  When I do, it discovers all 14 of my source files, but fails to load every one, indicating that I need to add them to other-modules or exposed-modules in the cabal file.  But they are already there.

The cabal file only has two stanzas:  a common and an executable.  I noticed that if I change the executable stanza into a library (and removed main-is), suddenly HLS is able to load the source files.

I also noticed that it works if I rename the executable stanza to VulkanSandboxExe, and then _add_ a library stanza to it, such as:
```
library
  import: common
```

Note: I tested and confirmed that renaming the executable stanza alone was not enough.

Did I miss something in the HLS documentation about requiring library stanzas?

**Comment by fendor** [comment]
Hi, thank you for your bug report!

I haven't checked in detail, but can you report what the behaviour is when you create a `hie.yaml` file with the contents:
```
cradle:
  cabal:
```

in the root of your project?

If this works, then the issue you are encountering is related to implicit-hie, which will be fixed in the next release. If it doesn't work, I will take another look :)

**Comment by Rotaerk** [comment]
Yep, adding `hie.yaml` containing the above to the same folder as my `VulkanSandbox.cabal` file (after having undone my change to add a library stanza to it) worked.  Thanks!
